### PR TITLE
class singleton created

### PR DESCRIPTION
**Pull request by brunomembrado**

### CODE REVIEW

File: src/lib/dbConnect.ts [hunk 1]
```diff
@@ -1,11 +1,49 @@
 import mongoose from "mongoose";
 
-async function dbConnect() {
+/* async function dbConnect() {
   try {
     await mongoose.connect(process.env.MONGODB_URI!);
   } catch (error) {
     throw new Error("Connection failed!");
   }
 }
```
Comment (solvd-interviews) [inline src/lib/dbConnect.ts]:
```suggestion
```

File: src/lib/dbConnect.ts [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
 
-export default dbConnect;
+export default dbConnect; */
```
Comment (solvd-interviews) [inline src/lib/dbConnect.ts]:
```suggestion
```

File: src/lib/dbConnect.ts [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
+    if (!DatabaseSingleton.instance) {
+      DatabaseSingleton.instance = new DatabaseSingleton();
+    }
+    return DatabaseSingleton.instance;
+  }
+
+  // Connect to the database if not already connected
+  public async connect() {
+    if (this.isConnected) {
+      console.log("Already connected to the database.");
```
Comment (andvirga) [inline src/lib/dbConnect.ts]: This will be printed for each connect() call, I think it's better to remove it
```suggestion
```

**Review by andvirga — APPROVED** [review]
LGTM, only left some minor suggestions (code cleanup)

**Pull request merged**